py2app/recipes/black.py

```diff
@@ -0,0 +1,25 @@
+from pkg_resources import get_distribution
+from pathlib import Path
+
+
+def check(cmd, mf):
+    m = mf.findNode("black")
+    if m is None or m.filename is None:
+        return None
+
+    egg = Path(get_distribution('black').egg_info)
+    top = egg / 'top_level.txt'
+
+    # These cannot be in zip
+    packages = ["black", "blib2to3"]
+
+    # black may include optimized platform specific C extension which has
+    # unusual name, e.g. 610faff656c4cfcbb4a3__mypyc; best to determine it from
+    # the egg-info/top_level.txt
+    includes = set(top.read_text().strip().split('\n'))
+    includes = list(includes.difference(packages))
+
+    # Missed dependency
+    includes.append('pathspec')
```

Comment:
I have to look into this a bit more, the problem here is that py2app performs a regular import when building the dependency graph, and hence picks up the C extension which cannot be scanned for dependencies.

This code is OK for now, but I'll probably research if I can detect if a package uses mypyc and try to scan the corresponding python sources in that case.   That should also take away the need for separate recipes for every package using mypyc.